pkgs/development/python-modules/wxPython/4.0.nix

```diff
@@ -22,14 +37,43 @@ buildPythonPackage rec {
     sha256 = "35cc8ae9dd5246e2c9861bb796026bbcb9fb083e4d49650f776622171ecdab37";
   };
 
-  nativeBuildInputs = [
-    pkgconfig
-  ];
+  doCheck = false;
 
-  buildInputs = [
-    gtk3 libjpeg libtiff SDL gst-plugins-base libnotify freeglut xorg.libSM
-    which
-  ];
+  nativeBuildInputs = [ pkgconfig which doxygen wxGTK libX11 ];
+
+  buildInputs = with python3Packages; [ libjpeg libtiff SDL
+      gst-plugins-base libnotify freeglut xorg.libSM ncurses
+      requests libpng gstreamer
+      pathlib2
+      (wxGTK.gtk)
+  ]
+    ++ (lib.optional openglSupport pyopengl);
+
+  hardeningDisable = [ "format" ];
+
+  DOXYGEN = "${doxygen}/bin/doxygen";
+
+  preConfigure = lib.optionalString (!stdenv.isDarwin) ''
+    substituteInPlace wx/lib/wxcairo/wx_pycairo.py \
+      --replace 'cairoLib = None' 'cairoLib = ctypes.CDLL("${cairo}/lib/libcairo.so")'
+    substituteInPlace wx/lib/wxcairo/wx_pycairo.py \
+      --replace '_dlls = dict()' '_dlls = {k: ctypes.CDLL(v) for k, v in [
+        ("gdk",        "${wxGTK.gtk}/lib/libgtk-x11-2.0.so"),
+        ("pangocairo", "${pango.out}/lib/libpangocairo-1.0.so"),
+        ("appsvc",     None)
+      ]}'
+  '';
+
+  buildPhase = ''
+  python build.py -v --use_syswx dox etg --nodoc sip build_py
+  '';
+
+  installPhase = ''
+    ${python3.interpreter} setup.py install --skip-build --prefix=$out
+    wrapPythonPrograms
```

Comment:
I think this is taken care of by the default pythonBuildPackage, should be able to remove your installPhase altogether.

Otherwise, there this is a buildPythonApplication; which may suit your needs better.

Reply:
@jonringer the --skip-build and --prefix are specific to wxPython. I tried installFlags but didn't get it working. Also, this is a module, not an application.

Reply:
more precisely, I tried `installFlags=["--install-option='--skip-build'"];` instead of installPhase

Reply:
hmmm, I see. I guess that makes sense then